src/pages/community/getting-help.mdx

```diff
@@ -6,33 +6,61 @@ How to engage with the Pants community.
 
 The [Pants community](/community/members) is friendly, welcoming, and passionate about improving the craft of software engineering. If you need help with anything Pants-related, you've come to the right place. We'd love to hear from you!
 
-## Slack
+## How to get help
 
-The best place to become acquainted with Pants community members, exchange informal chat and ask quick questions is at the project's Slack. You can join using [this link](https://docs.google.com/forms/d/e/1FAIpQLSf9zgf-MXRnVDJbrVEST3urqneq7LCcy0zw8qU-GH4hPMn52A/viewform?usp=sf_link).
+Pants community members will do their best to answer any questions that are posed in a respectful way. General questions about Pants,
+its features, language support, roadmap etc. are always welcome.
+
+If you need help debugging a specific problem you're encountering in your repo, then we ask that you help us to help you by adhering to the following:
+
+- First, do your best to solve the issue on your own. The Pants [source code](https://github.com/pantsbuild/pants), [issue tracker](https://github.com/pantsbuild/pants/issues), and [documentation](https://pantsbuild.org)
+  are useful resources for this.
+- If you're still stuck, and would like a Pants contributor to look at your problem, please provide a small public GitHub repo that reproduces the problem.
+  The README.md in this repo should provide clear instructions on how to reproduce the problem, and the architecture and OS on which it can be reproduced.
+  Posting code and config snippets to Slack is not a good alternative, as we won't know if they contain enough information to reproduce the problem.
+  A self-contained repo is far more useful.
+- Then [file an issue](https://github.com/pantsbuild/pants/issues/new/choose) that states the problem and the steps you've taken to attempt to solve it, and a link to the reproduction repo. If you find
+  an existing issue that seems relevant, please link to it, or post directly to that existing issue if you're reasonably sure it describes the same problem.
+- Then, feel free to reach out in the #general channel in Slack with a link to your issue, to kickstart the discussion.
+
+Keep in mind that this community support is provided _voluntarily_. When you ask for help you're asking busy people to devote their valuable time to solving your problem.
+You are far more likely to get a positive response if you demonstrate that you've put in the basic legwork first.
```

Comment:
I strongly appreciate encouraging people to provide the right info straight up, but I'm a little concerned about phrasing some of this stuff too strongly in terms of "you must do all these things [or else you'll be ignored for process reasons]" which seems... scary, especially for someone new to open source.

I'd personally prefer to phrase it in terms of what's in it for the asker, i.e. focus on "help us help you" / "you'll fix your problem quicker if you make it easy for us to help".

For instance:

```suggestion
You are far more likely to get a positive response if you make it easy for people to help you.
```

But I think potentially this whole section could be reframed slightly to be cast in that light, especially the list above.

What do you think?

Reply:
Are there any projects that do this kind of language well (friendly, while direct and helpful?).

I'm always a fan of how Svelte phrases things, for example.

![Screenshot 2024-03-07 at 10 09 38](https://github.com/pantsbuild/pantsbuild.org/assets/3823045/c60827e6-2907-4a2e-aef2-186f2dc8553c)

Reply:
> I'm always a fan of how Svelte phrases things, for example.

Ah yeah, I like it too.

Somewhat tangential but that text there makes me wonder if we could also update our bug report template to be more specific about what's expected.

>  Are there any projects that do this kind of language well (friendly, while direct and helpful?).

Good question! I don't know of any particularly good examples of the top of my head.


I know Rust (used to) put a lot of effort into helping new-comers, and this is their current "generic bug" issue template:

``````markdown
<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
<code>
```

I expected to see this happen: *explanation*

Instead, this happened: *explanation*

### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
<version>
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>
``````

Reply:
I'll just take this bit out.